backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUPIItemProductBL.java

```diff
@@ -195,4 +203,49 @@ public I_M_HU_PI_Item_Product getDefaultForProduct(@NonNull final ProductId prod
 	{
 		return huPIItemProductDAO.retrieveDefaultForProduct(productId, dateTime);
 	}
+
+	@NonNull
+	public I_M_HU_PI_Item_Product extractHUPIItemProduct(final I_C_Order order, final I_C_OrderLine orderLine)
+	{
+		final I_M_HU_PI_Item_Product materialItemProduct;
+		if (orderLine.getM_HU_PI_Item_Product_ID() > 0)
+		{
+			materialItemProduct = orderLine.getM_HU_PI_Item_Product();
+		}
+		else
+		{
+			final ProductId productId = ProductId.ofRepoId(orderLine.getM_Product_ID());
+			final BPartnerId buyerBPartnerId = BPartnerId.ofRepoId(order.getC_BPartner_ID());
+			final ZoneId timeZone = orgDAO.getTimeZone(OrgId.ofRepoId(order.getAD_Org_ID()));
+
+			materialItemProduct = huPIItemProductDAO.retrieveMaterialItemProduct(
+					productId,
+					buyerBPartnerId,
+					TimeUtil.asZonedDateTime(order.getDateOrdered(), timeZone),
+					X_M_HU_PI_Version.HU_UNITTYPE_TransportUnit, true/* allowInfiniteCapacity */);
+		}
+
+		return materialItemProduct;
+	}
+
+	@Override
+	public int getRequiredLUCount(final @NonNull StockQtyAndUOMQty qtyToAdd, final I_M_HU_LUTU_Configuration lutuConfigurationInStockUOM)
+	{
+		final int requiredLUCount;
```

Comment:
don't need this local variable, simply return on the spot

Reply:
returning directly.